.docker/wp-config.php

```diff
@@ -1,6 +1,6 @@
 <?php
 
-define( 'WORDCAMP_ENVIRONMENT', 'development' );
+define( 'WORDCAMP_ENVIRONMENT', 'local' );
```

Comment:
Will this affect other things using `WORDCAMP_ENVIRONMENT`? A quick look seems to show that most conditionals are checking for `production` and fall back to dev settings in all other cases, so it might be fine.

Reply:
Yeah, as far as I can tell there isn't any code that is specifically looking for `development` in that constant.

Reply:
Here are the specific things I think might be good to change, if my assumptions are correct. Or at the very least maybe this will help reveal where we might have different assumptions.

`SHOULD` - If we add `local`, I think we should also change the remote sandbox value to be `staging` or `remote`. `development` seems like it'd be easily confused in that context, since conventionally that means a local environment in my experience.

- [ ] `MUST`: It seems like the `wcorg_sponsor_payment_stripe` and `wordcamp_google_maps_api_key` closures in `wcorg-misc.php` would need to be updated, otherwise the keys won't be populated because there's no `local` case in the `switch`.
- [ ] `MUST`: `wcpt-event/notification.php` should probably return early on local sandboxes instead of remote.

`MIGHT` - Would it be simpler to add `case 'local'` to `set_client_config()` in `quickbooks.php`, instead of devs needing to manually add a new `wordcamp_qbo_client_config` callback in `sandbox-functionality.php`?

`MIGHT` - Right now `set_client_config()` is using the sandbox keys in local and remote sandboxes, but it might make more sense to use the production keys in `staging`/`remote` and the sandbox keys in `local`.

Related in Trello: `/c/rErUVQ6P` (first 2 uncompleted checklist items) and `/c/5exJNacO`

Reply:
> What problems did you run into while testing locally that prompted this?

The problem was that there were several places in the code that had to be temporarily modified in order to be able to workflows involving QBO (hardcoded API URLs, the SSL verification setting, etc). This meant leaving the working directory dirty and introducing the risk of accidentally committing these temporary changes.

> What's the reason for introducing a distinction between local and remote sandboxes?

In a local dev environment, the domain is **wordcamp.test**. In both production and on sandboxes, the domain is **wordcamp.org**. There are other differences as well, such as that sandboxes can have two-way communication with other remote servers (e.g. Jetpack connection stuff), while local environments can't (unless you use ngrok or something)

Reply:
> If we add local, I think we should also change the remote sandbox value to be staging or remote.

Ok, although I don't think that needs to be part of this PR, since this is specifically about making it easier to test QBO locally.

> It seems like the wcorg_sponsor_payment_stripe and wordcamp_google_maps_api_key closures in wcorg-misc.php would need to be updated, otherwise the keys won't be populated because there's no local case in the switch.

In both of those closures, the `default` switch case catches "local", so it gets the same treatment as if it was "development".

> wcpt-event/notification.php should probably return early on local sandboxes instead of remote.

Good call. I guess we could just change the line at the top to

```php
if ( 'local' === WORDCAMP_ENVIRONMENT ) {
	return;
}
```

> Would it be simpler to add case 'local' to set_client_config() in quickbooks.php, instead of devs needing to manually add a new wordcamp_qbo_client_config callback in sandbox-functionality.php?

We don't want to publish our QBO sandbox keys in the repo, so the values would need to go in **sandbox-functionality.php** either way. Otherwise you end up putting those values in a file under version control and risk accidentally committing them.

> Right now set_client_config() is using the sandbox keys in local and remote sandboxes, but it might make more sense to use the production keys in staging/remote and the sandbox keys in local.

I think remote sandboxes should still use sandbox keys. If we're testing interactions with the QBO API on a sandbox, we wouldn't want test data to get mixed in with actual transactions.

Reply:
Re: this comment in Trello:

> The various usages of `WORDCAMP_ENVIRONMENT` throughout the codebase are starting to feel kind of `WET`, because they're not assuming that it exists. It does (and should/must) exist in all of our environments, though, so that feels like it's adding a lot of needless complexity. What about just removing all `defined()` calls and fallbacks etc, and just referencing the value directly everywhere?

The `get_wordcamp_environment()` function introduced in this PR is meant to address this. I didn't go through and replace all the references to the constant in the whole codebase, because that seemed out of scope, but we could easily do that in a separate PR once this is merged.

Reply:
> wcpt-event/notification.php should probably return early on local sandboxes instead of remote.

1bb2e2eca0ba693df20964dd844375e344652835

Reply:
> I don't think [changing `development` to `staging`] needs to be part of this PR

I didn't realize you were seeing environments being [_both_ `local` and `development` simultaneously](
https://github.com/WordPress/wordcamp.org/pull/371#discussion_r381501047), rather than changing `development` to _only_ mean remote sandboxes. So I guess in that case it shouldn't change 👍 



> In both of those closures, the default switch case catches "local", so it gets the same treatment as if it was "development".

Ah, you're right, I missed that. 👍 


> just change the line at the top to...

👍 


> the values would need to go in sandbox-functionality.php either way. Otherwise [...] version control

Ah, I didn't realize that your `public_html/wp-config.php` gets overwritten when you re-provision. 👍 


> I think remote sandboxes should still use sandbox keys. If we're testing interactions with the QBO API on a sandbox, we wouldn't want test data to get mixed in with actual transactions.

I guess we just use remote sandboxes differently. I only use mine as a staging environment, like trying to reproduce a bug someone reported w/ production data. Either way works 👍